front/src/common/Map/Search/MapSearchSignal.tsx

```diff
@@ -0,0 +1,272 @@
+import { getInfraID } from 'reducers/osrdconf/selectors';
+import { setFailure } from 'reducers/main';
+import { sortBy } from 'lodash';
+import { ISignalSearchResult } from './searchTypes';
+import { searchPayloadType, signalAspects } from '../const';
+import SignalCard from './SignalCard';
+import { onResultSearchClick } from '../utils';
+
+type MapSearchSignalProps = {
+  updateExtViewport: (viewport: Partial<Viewport>) => void;
+};
+
+export type SortType = {
+  name: 'label' | 'type' | 'line_name' | 'line_code';
+  asc: boolean;
+};
+
+const SIGNAL_ASPECTS = signalAspects;
+
+const MapSearchSignal = ({ updateExtViewport }: MapSearchSignalProps) => {
+  const map = useSelector(getMap);
+  const infraID = useSelector(getInfraID);
+
+  const [searchState, setSearch] = useState('');
+  const [searchLineState, setSearchLine] = useState('');
+  const [aspects, setAspects] = useState<string[]>([]);
+  const [searchResults, setSearchResults] = useState<ISignalSearchResult[]>([]);
+  const [autocompleteLineNames, setAutocompleteLineNames] = useState<string[]>([]);
+
+  // Sort by, and order true = ASC, false = DESC
+  const [sortFilter, setSortFilter] = useState<SortType>({
+    name: 'label',
+    asc: true,
+  });
+  const dispatch = useDispatch();
+  const [postSearch] = osrdEditoastApi.usePostSearchMutation();
+  const { t } = useTranslation(['translation', 'map-search']);
+
+  const getPayload = (
+    lineSearch: string,
+    signalName: string,
+    infraIDPayload: number
+  ): searchPayloadType => {
+    const payloadQuery = !Number.isNaN(Number(lineSearch))
+      ? ['=', ['line_code'], Number(lineSearch)]
+      : ['search', ['line_name'], lineSearch];
+
+    return {
+      object: 'signal',
+      query: [
+        'and',
+        ['=', ['infra_id'], infraIDPayload],
+        !lineSearch || payloadQuery,
+        !aspects.length || ['contains', ['list', ...aspects], ['aspects']],
+        ['search', ['label'], signalName],
+      ],
+    };
+  };
+
+  const updateSearch = async (infraIDPayload: number) => {
+    const payload: searchPayloadType = getPayload(searchLineState, searchState, infraIDPayload);
+    await postSearch({
+      body: payload,
+    })
+      .unwrap()
+      .then((results) => {
+        setSearchResults([...results] as ISignalSearchResult[]);
+      })
+      .catch((e) => {
+        console.error(e);
+        setSearchResults([]);
+        dispatch(
+          setFailure({
+            name: t('map-search:errorMessages.unableToSearchSignal'),
+            message: `${e.message}`,
+          })
+        );
+      });
+  };
+
+  const debouncedSearchTerm = useDebounce(searchState, 300);
+  const debouncedSearchLine = useDebounce(searchLineState, 300);
+
+  useEffect(() => {
+    if ((searchLineState || searchState) && infraID) {
+      updateSearch(infraID);
+    } else {
+      setSearchResults([]);
+    }
+  }, [debouncedSearchTerm, debouncedSearchLine, aspects, searchState]);
```

Comment:
```suggestion
  }, [debouncedSearchTerm, debouncedSearchLine, aspects]);
```
No need for searchState here, as we already have debouncedSearchTerm

Reply:
Done !!